golos.publication/golos.publication.cpp

```diff
@@ -122,8 +122,19 @@ void publication::create_message(account_name account, std::string permlink,
         item.tags = tags;
         item.jsonmetadata = jsonmetadata;
     });
-
-    close_message_timer(account, message_id);
+    
+    structures::accandvalue parent {parentacc, parent_id};
+    uint64_t date_diff = 0;
+    while (parent.account) {
+        tables::message_table parent_message_table(_self, parent.account);
+        auto parent_itr = parent_message_table.find(parent.value);
+        eosio_assert(cur_time >= parent_itr->date, "publication::create_message: cur_time < parent_itr->date");
```

Comment:
how it's possible? it's good to write test to check each assert. if you can't do this, an assert is not needed, it only can occur on totally broken system, and if so, it will not help here

Reply:
in the case of a “completely broken system,” it would be handy to see the nearest point where something went wrong. Should we have maybe some sort of logic_error_assert for such situations?

Reply:
actually, I had similar idea: to have some special assert type for such checks. agree, it can be useful